packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts

```diff
@@ -1464,32 +1503,34 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 
 		// pause outgoing ops so we can detect dropped stashed changes
 		await container.deltaManager.outbound.pause();
-
-		let pendingState;
-		// always delete stash blob on "connected" event
-		container.on("connected", (clientId: string) => {
-			pendingState = container.getPendingLocalState?.();
-			// the pending data in the stash blob may not have changed, but the clientId should match our new
-			// clientId, which will now be used to attempt to resubmit pending changes
-			assert.strictEqual(clientId, JSON.parse(pendingState).clientId);
-		});
-
+		let pendingState: string | undefined;
+		let pendingStateP;
 		const dataStore = await requestFluidObject<ITestFluidObject>(container, "default");
 		const map = await dataStore.getSharedObject<SharedMap>(mapId);
 		for (let i = 5; i--; ) {
 			map.set(`${i}`, `${i}`);
 			container.disconnect();
 			container.connect();
-			await new Promise<void>((resolve) => container.on("connected", () => resolve()));
+			pendingStateP = await new Promise<string>((resolve) => {
+				container.once("connected", (clientId: string) => resolve(clientId));
```

Comment:
I think we are losing something by refactoring this way, since this no longer works as an example of how to use this API safely. Since there isn't always an event listener clearing stash blobs on "connected" events, this could still result in stale stash blobs and duplicate op submission. In real usage, the host will not control all connect/disconnects.

Reply:
I'll fix this in a follow up